### PR TITLE
calculate reward on stake

### PR DESCRIPTION
**Pull request by blasrodri**

### CODE REVIEW

File: programs/candidate_staking/src/lib.rs [hunk 1]
```diff
@@ -144,9 +143,11 @@ pub mod candidate_staking {
                     outer.as_slice(), //signer PDA
                 );
 
+                let amount_in_32 = ctx.accounts.base_account.reward_amount as u64;
```
Comment (blasrodri) [inline programs/candidate_staking/src/lib.rs:146]:
```suggestion
                let amount_in_64 = ctx.accounts.base_account.reward_amount as u64;
```

File: programs/candidate_staking/src/lib.rs [hunk 2]
```diff
@@ -144,9 +143,11 @@ pub mod candidate_staking {
...
+
                 // The `?` at the end will cause the function to return early in case of an error.
                 // This pattern is common in Rust.
-                anchor_spl::token::transfer(cpi_ctx, reward)?;
+                anchor_spl::token::transfer(cpi_ctx, amount_in_32)?;
```
Comment (blasrodri) [inline programs/candidate_staking/src/lib.rs:150]:
```suggestion
                anchor_spl::token::transfer(cpi_ctx, amount_in_64)?;
```

Comment (dhruvja) [reply programs/candidate_staking/src/lib.rs:150]: noted

File: programs/candidate_staking/src/lib.rs (hunk 1 above)
Comment (dhruvja) [reply programs/candidate_staking/src/lib.rs:146]: noted

**Review by dhruvja — COMMENTED** [review]
i have pushed those changes. The rewards based on tiers are working and all the test cases are passing. But we need to write some more test cases.

**Comment by blasrodri** [comment]
I've rebased master, so now we get to see the output of `anchor test`

**Pull request merged**